definitions/tools/replace_vcf_sample_name.cwl

```diff
@@ -18,7 +18,13 @@ requirements:
           set -eou pipefail
           basen=`basename "$3"`
           basen="renamed.$basen"
-          echo "$1 $2" > sample_update.txt
+
+          #escape spaces, otherwise bcftools will try to use them as a delimiter
+          #triple backslash to escape within backticks and then again within sed
+          old_name=`echo "$1" | sed 's/ /\\\ /g'`
+          new_name=`echo "$2" | sed 's/ /\\\ /g'`
+
+          echo "$old_name $new_name" > sample_update.txt
```

Comment:
Part of me wants to suggest switching to the one-sample per line format, but that'd require this tool to handle all the samples simultaneously.  Got to love custom space-delimited formats!

Reply:
For future reference, this is documented [in the bcftools docs](https://samtools.github.io/bcftools/bcftools.html#reheader). I agree that the sample per line format would have been easier to use, but as you mentioned, I'd then also have to handle all samples at once, and figure out what order they're recorded in the vcf to do this properly. This way is a little more of a hack, but safer for our use case.